### ISSUE TITLE
examples: Timelock program

### ISSUE THREAD

**Issue by armaniferrante** [solana-foundation/anchor#91]
A useful example would be a timelock program that executes an arbitrary transaction at a specified point in time in the future.

**Comment by jon-chuang** [comment]
This is interesting! Should it use `sysvar::clock()`?

**Comment by armaniferrante** [comment]
Correct. One would need to trigger the program externally with a permissionless transaction. Even so, this is still very usefull. For example, one compelling use case is for one to compose a [multisig](https://github.com/project-serum/multisig) program with a timelock program to perform program upgrades.

**Comment by jon-chuang** [comment]
Hi @armaniferrante , I've recently shown how multisig program taken to its generic conclusion is equivalent to a pupeteer which can also schedule transactions at arbitrary times or recur them indefinitely.

You can take a look here: https://github.com/solana-labs/solana/issues/17218

**Comment by WrRaThY** [comment]
@jon-chuang - do you have any implementation of this?
I will need something like that in my program ;)
